### ISSUE TITLE
Noam learning schedule causes gradient vanishing problem

### ISSUE THREAD

**Issue by emremidilli** [emremidilli/Time-Series-Forecasting#9]
Default warmup steps are set as 4000. Aprx. after 3000, metrics suddenly drop significantly. Later it improves the reduces the loss function with very small amounts.  This is a clear sign of gradient vanishing problem.

**Comment by emremidilli** [comment]
We may consider to re-set the warmup_steps property of the callback object in case there is a significant drop on metrics during training. Also, we must have separate learning rates for masked-autoencoder and contrastive learning tasks.

**Comment by emremidilli** [comment]
Reducing the warmup steps dont work much. Main reason of gradient vanishing is high learning rate. Main reason of high learning rate is d_model parameter of the scheduler.

**Comment by emremidilli** [comment]
With scale_factor parameter, learning rate is decreased. In case gradient vanishing occurs during trial runs, just decrease the learning scale_factor.
